src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joined = [...this];
```

Comment:
It's not something joined. In this variable you've stored the copy of original array